recipe/build.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -ex
+
+export RUST_LOG=debug
+export RUST_BACKTRACE=full
+
+# See https://github.com/conda-forge/rust-feedstock/blob/master/recipe/build.sh for cc env explanation
+
+if [ "$c_compiler" = gcc ] ; then
+    case "$target_platform" in
+        linux-64) rust_env_arch=X86_64_UNKNOWN_LINUX_GNU ;;
+        linux-aarch64) rust_env_arch=AARCH64_UNKNOWN_LINUX_GNU ;;
+        linux-ppc64le) rust_env_arch=POWERPC64LE_UNKNOWN_LINUX_GNU ;;
+        *) echo "unknown target_platform $target_platform" ; exit 1 ;;
+    esac
+
+    export CARGO_TARGET_${rust_env_arch}_LINKER=$CC
+fi
+
+declare -a _xtra_maturin_args
+#_xtra_maturin_args+=(--cargo-extra-args="-Zfeatures=itarget")
+# _xtra_maturin_args+=(-Zfeatures=itarget)
+
+if [ "$target_platform" = "osx-64" ] || [ "$target_platform" = "osx-arm64" ] ; then
+    # This variable must be set to the directory containing the target's libpython DSO
+    export PYO3_CROSS_LIB_DIR=$PREFIX/lib
+
+    # xref: https://github.com/PyO3/pyo3/commit/7beb2720
+    export PYO3_PYTHON_VERSION=${PY_VER}
+
+    mkdir -p $SRC_DIR/.cargo
+    cat <<EOF >> $SRC_DIR/.cargo/config
+# Required for intermediate codegen stuff
+[target.x86_64-apple-darwin]
+linker = "$CC_FOR_BUILD"
+
+# Required for final binary artifacts for target
+[target.aarch64-apple-darwin]
+linker = "$CC"
+rustflags = [
+  "-C", "link-arg=-undefined",
+  "-C", "link-arg=dynamic_lookup",
+]
+
+EOF
+    if [ "$target_platform" = "osx-64" ] ; then
+        _xtra_maturin_args+=(--target=x86_64-apple-darwin)
+    else
+        _xtra_maturin_args+=(--target=aarch64-apple-darwin)
+    fi
+
+    # xref: https://github.com/conda-forge/python-feedstock/issues/621
+    sed -i.bak 's,aarch64,arm64,g' $BUILD_PREFIX/venv/lib/os-patch.py
```

Comment:
Looks like this file doesn't exit on macOS x86_64. From [CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=798559&view=logs&j=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&t=20c71c51-4b27-578b-485d-06ade2de1d00&l=192 ):

```
+ sed -i.bak s,aarch64,arm64,g /Users/runner/miniforge3/conda-bld/dask-sql_1696630065483/_build_env/venv/lib/os-patch.py
sed: /Users/runner/miniforge3/conda-bld/dask-sql_1696630065483/_build_env/venv/lib/os-patch.py: No such file or directory
```

<br>

Should we add a conditional around this?